src/app/core-ui/main/release-notification/release-notification.component.ts

```diff
@@ -15,15 +16,15 @@ export class ReleaseNotificationComponent implements OnInit {
   public latestClientVersion: string;
   public releaseUrl: string;
 
-  constructor(private http: HttpClient) { }
+  constructor(private clientVersionService: ClientVersionService) { }
 
   ngOnInit() {
     // check new update in every 30 minute
     setTimeout(this.getCurrentClientVersion(), 1800000);
```

Comment:
This will check for updates once and then stop?

Maybe use Observables and the interval(), and make sure to destroy the subscription through takeWhile() ngOnDestroy.

Reply:
@kewde no it will get client version in every 30 min, and yes we can destroy `setTimeout` if a new version gets. but no need I guess!!

Reply:
@anandsinghparihar 

No,
setTimeout runs once. setInterval is the right one to use here?
You don't understand tho, this code will keep on running, even if the component is destroyed.
Just like we do with takeWhile(() => !this.destroyed), you need to stop setInterval() and clean it up if the component has been destroyed (use ngOnDestroy).

Reply:
https://stackoverflow.com/questions/37116619/angular-2-setinterval-keep-running-on-other-component

Reply:
@kewde Yup got it, changes pushed